src/components/ArmyBuildPane.jsx

```diff
@@ -17,30 +17,42 @@ const Container = styled.div`
 `;
 
 const ArmyBuildPane = () => {
+
   const [unitList, setUnitList] = useState({});
   const [listIndex, setListIndex] = useState(1);
 
   const addUnit = unit => {
+    unit.selected_upgrades = {};
```

Comment:
I suspect this won't be the only property you'll want to initialize with a blank value. You could create an object of named properties with their default values and spread it on the unit so that you just have a bunch of hanging properties in this function. 

So, slight tweak on joshs comment:

```
[listIndex]: {
    ...unit,
    ...baseProperties //insert better name here
}
```